File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-gitee/src/main/java/com/tencent/devops/scm/provider/git/gitee/auth/GiteeTokenAuthProviderAdapter.java
package com.tencent.devops.scm.provider.git.gitee.auth;

import com.tencent.devops.scm.api.pojo.auth.AccessTokenScmAuth;
import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.api.pojo.auth.PersonalAccessTokenScmAuth;
import com.tencent.devops.scm.api.pojo.auth.TokenSshPrivateKeyScmAuth;
import com.tencent.devops.scm.api.pojo.auth.TokenUserPassScmAuth;
import com.tencent.devops.scm.sdk.gitee.auth.GiteeTokenAuthProvider;

public class GiteeTokenAuthProviderAdapter {

    public static boolean support(IScmAuth auth) {
        return auth instanceof AccessTokenScmAuth
                || auth instanceof PersonalAccessTokenScmAuth
                || auth instanceof TokenUserPassScmAuth
                || auth instanceof TokenSshPrivateKeyScmAuth;
    }

    public static GiteeTokenAuthProvider get(IScmAuth auth) {
        if (auth instanceof AccessTokenScmAuth) {
            return GiteeTokenAuthProvider.fromOauthToken(((AccessTokenScmAuth) auth).getAccessToken());
        }
        if (auth instanceof PersonalAccessTokenScmAuth) {
            return GiteeTokenAuthProvider.fromPersonalAccessToken(
                    ((PersonalAccessTokenScmAuth) auth).getPersonalAccessToken());
        }
        if (auth instanceof TokenUserPassScmAuth) {
            return GiteeTokenAuthProvider.fromPersonalAccessToken(((TokenUserPassScmAuth) auth).getToken());
        }
        if (auth instanceof TokenSshPrivateKeyScmAuth) {
            return GiteeTokenAuthProvider.fromPersonalAccessToken(((TokenSshPrivateKeyScmAuth) auth).getToken());
        }
        throw new UnsupportedOperationException(String.format("gitAuth(%s) is not support", auth));
    }
}

File: devops-scm-provider/devops-scm-provider-git/devops-scm-provider-gitee/src/main/kotlin/com/tencent/devops/scm/provider/git/gitee/auth/GiteeTokenAuthProviderAdapter.kt
package com.tencent.devops.scm.provider.git.gitee.auth

import com.tencent.devops.scm.api.pojo.auth.AccessTokenScmAuth
import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.api.pojo.auth.PersonalAccessTokenScmAuth
import com.tencent.devops.scm.api.pojo.auth.TokenSshPrivateKeyScmAuth
import com.tencent.devops.scm.api.pojo.auth.TokenUserPassScmAuth
import com.tencent.devops.scm.sdk.gitee.GiteeConstants.TokenType
import com.tencent.devops.scm.sdk.gitee.auth.GiteeTokenAuthProvider

object GiteeTokenAuthProviderAdapter {

    fun support(auth: IScmAuth): Boolean {
        return auth is AccessTokenScmAuth ||
                auth is PersonalAccessTokenScmAuth ||
                auth is TokenUserPassScmAuth ||
                auth is TokenSshPrivateKeyScmAuth
    }

    fun get(auth: IScmAuth): GiteeTokenAuthProvider {
        return when (auth) {
            is AccessTokenScmAuth -> GiteeTokenAuthProvider.fromTokenType(
                TokenType.OAUTH2_ACCESS,
                auth.accessToken
            )
            
            is PersonalAccessTokenScmAuth -> GiteeTokenAuthProvider.fromTokenType(
                TokenType.PERSONAL_ACCESS,
                auth.personalAccessToken
            )
            
            is TokenUserPassScmAuth -> GiteeTokenAuthProvider.fromTokenType(
                TokenType.PERSONAL_ACCESS,
                auth.token
            )
            
            is TokenSshPrivateKeyScmAuth -> GiteeTokenAuthProvider.fromTokenType(
                TokenType.PERSONAL_ACCESS,
                auth.token
            )
            
            else -> throw UnsupportedOperationException("gitAuth($auth) is not support")
        }
    }
}
